src/app/confirm-call-destination/_components/ActionButton.tsx

```diff
@@ -2,36 +2,51 @@
 
 import { Button, Flex } from '@kuma-ui/core';
 import { useRouter } from 'next/navigation';
+import { NextResponse } from 'next/server';
+
+const ENDPOINT_BASE_PATH = process.env.NEXT_PUBLIC_ENDPOINT_BASE_PATH
 
 export const GeneralReceptionActionButtons = () => {
   const handleCancel = useActionButton()
-  const handleConfirm: React.ComponentProps<typeof ActionButtons>['onConfirm'] = (e) => {
-    e.preventDefault()
-    alert('総合受付')
-  }
+  const handleConfirm = usesendMessage('general-reception')
   return <ActionButtons onConfirm={handleConfirm} onCancel={handleCancel} />
 }
 
 export const RecruitmentInterviewActionButtons = () => {
   const handleCancel = useActionButton()
-  const handleConfirm: React.ComponentProps<typeof ActionButtons>['onConfirm'] = (e) => {
-    e.preventDefault()
-    alert('採用面接')
-  }
+  const handleConfirm = usesendMessage('recruitment-interview')
   return <ActionButtons onConfirm={handleConfirm} onCancel={handleCancel} />
 }
 
-const useActionButton = () => {
+const useActionButton = () =>{
   const router = useRouter();
-
   const handleCancel: React.ComponentProps<typeof ActionButtons>['onCancel'] = (e) => {
     e.preventDefault();
     router.push('/');
   };
-
   return handleCancel
 }
 
+const usesendMessage = (endpoint:string) =>{
+  const sendMessage: React.ComponentProps<typeof ActionButtons>['onConfirm'] = async(e) => {
+    e.preventDefault();
+    const response = await fetch(`${ENDPOINT_BASE_PATH}/${endpoint}`, { //バックエンドデプロイ後正しいURLに変更
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    })
+
+    if (!response.ok) {
+      return NextResponse.json({ error: 'Failed to send message' }, { status: response.status })
+    }
+
+    const data = await response.json()
+    return NextResponse.json(data)
```

Comment:
returnする必要ある？

```suggestion
```

Reply:
41行目は何となくここでreturnした値をrouter.push等で呼び出し失敗画面にエラーメッセージとしてアラートで表示しようと考えていてreturnしていましたが、それが必要なければreturnする必要もないと思います。
45行目はレスポンスが200なのか401なのかによって呼び出し完了画面と呼び出し失敗画面のどちらに遷移するかを決めようと思っているためreturnするべきかと考えていました。

Reply:
OKです。そしたら書きたいです。
また、「エラーメッセージとしてアラートで表示しようと考えていて」や「レスポンスが200なのか401なのかによって呼び出し完了画面と呼び出し失敗画面のどちらに遷移するかを決めよう」はコードにコメントして残しておいて欲しいです。

```suggestion
    if (!response.ok) {
      return { error: 'Failed to send message' , status: response.status }
    }

    const data = await response.json()
    return data
```

Reply:
承知しました！申し訳ありません。